repositories/rosbag2.BUILD.bazel

```diff
@@ -32,28 +33,78 @@ ros2_cpp_library(
     ],
 )
 
-ros2_plugin(
+cc_library(
     name = "rosbag2_storage_default_plugins",
     srcs = glob([
         "rosbag2_storage_default_plugins/src/**/*.cpp",
         "rosbag2_storage_default_plugins/src/**/*.hpp",
     ]),
     hdrs = glob(["rosbag2_storage_default_plugins/include/**/*.hpp"]),
     includes = ["rosbag2_storage_default_plugins/include"],
+    deps = [
+        ":rosbag2_storage",
+        "@ros2_pluginlib//:pluginlib",
+        "@ros2_rcpputils//:rcpputils",
+        "@ros2_rcutils//:rcutils",
+        "@sqlite",
+        "@yaml_cpp",
+    ],
+    alwayslink = True,
+)
+
+cc_library(
+    name = "mcap_vendor",
+    srcs = ["mcap_vendor/src/main.cpp"],
+    deps = [
+        "@mcap",
+        "@zstd",
+    ],
+)
+
+cc_library(
+    name = "rosbag2_storage_mcap",
+    srcs = [
+        "rosbag2_storage_mcap/src/mcap_storage.cpp",
+        "rosbag2_storage_mcap/src/message_definition_cache.cpp",
+    ],
+    hdrs = [
+        "rosbag2_storage_mcap/include/rosbag2_storage_mcap/message_definition_cache.hpp",
+        "rosbag2_storage_mcap/include/rosbag2_storage_mcap/visibility_control.hpp",
+    ],
+    defines = [
+        "ROSBAG2_STORAGE_MCAP_HAS_YAML_HPP",
+        "ROSBAG2_STORAGE_MCAP_HAS_STORAGE_OPTIONS",
+        "ROSBAG2_STORAGE_MCAP_WRITER_CREATES_DIRECTORY",
+        "ROSBAG2_STORAGE_MCAP_OVERRIDE_SEEK_METHOD",
+    ],
+    includes = ["rosbag2_storage_mcap/include"],
+    deps = [
+        ":mcap_vendor",
+        ":rosbag2_storage",
+        "@ros2_pluginlib//:pluginlib",
+        "@ros2_rcpputils//:rcpputils",
+        "@ros2_rcutils//:rcutils",
+    ],
+    alwayslink = True,
+)
+
+ros2_plugin(
```

Comment:
OK, I see what you want to achieve. Having pattern where cc_library the only has deps should be avoided IMO. Mainly because people tend to add deps and forget to remove them. We'll document what plugins are available and how to include extra plugins. We can always include the default one, and users will know how to include additional ones if they want.